src/comparison_tools/PairwiseCompare.py

```diff
@@ -212,7 +212,7 @@ def inter_comparisons(self):
             # Iterate through each well group cartesian product and save the data
             for ppair in comparison_key_product:
 
-                if self.__contains_match(ppair):
+                if self.__contains_match(ppair, self.__posthoc_group_cols):
```

Comment:
Consider adding comments here and throughout to help detail these if-scenarios and what this does.